### PR TITLE
Fix #85 Drag bug in Firefox

### LINKED ISSUE #85: Drag bug in Firefox
Version 3.0.0

Tested in Firefox 96.0.
Chrome seems to work differently and doesn't show the same bug.

drag_left_bug

### Steps to reproduce

1. Run up the demo page in Firefox.
2. Click the bottom image (Drag Example)
3. Drag to the very left of the screen.
4. Note the exception in the console.

### Exception

```
Uncaught TypeError: e.changedTouches is undefined
    handleDragMove InnerImageZoom.js:122
    InnerImageZoom InnerImageZoom.js:291
    React 6
    unstable_runWithPriority scheduler.development.js:653
    React 3
    workLoop scheduler.development.js:597
    flushWork scheduler.development.js:552
    performWorkUntilDeadline scheduler.development.js:164
    js scheduler.development.js:187
    js scheduler.development.js:857
    Webpack 21
InnerImageZoom.js:122
```

### Root Cause

In the code below e.pageX has the value 0 when the mouse is dragged to the left side of the window.
This is falsy and so e.changedTouches[0].pageX is executed instead and throws the exception because e.changedTouches does not exist.

```
  const handleDragMove = useCallback((e) => {
    e.stopPropagation();
    let left = (e.pageX || e.changedTouches[0].pageX) - imgProps.current.offsets.x;
    let top = (e.pageY || e.changedTouches[0].pageY) - imgProps.current.offsets.y;

    left = Math.max(Math.min(left, 0), (imgProps.current.scaledDimensions.width - imgProps.current.bounds.width) * -1);
    top = Math.max(Math.min(top, 0), (imgProps.current.scaledDimensions.height - imgProps.current.bounds.height) * -1);

    setLeft(left);
    setTop(top);
  }, []);
```

### Test Fix

This removes the exception.
If debugging the console.log will print out X and Y values.

```
  const handleDragMove = useCallback((e) => {
    e.stopPropagation();
    const pageX = typeof e.pageX === 'number' ? e.pageX : e.changedTouches[0].pageX;
    const pageY = typeof e.pageY === 'number' ? e.pageY : e.changedTouches[0].pageY;
    //console.log(`Move pageX: ${pageX} pageY: ${pageY}`);
    let left = pageX - imgProps.current.offsets.x;
    let top = pageY - imgProps.current.offsets.y;

    left = Math.max(Math.min(left, 0), (imgProps.current.scaledDimensions.width - imgProps.current.bounds.width) * -1);
    top = Math.max(Math.min(top, 0), (imgProps.current.scaledDimensions.height - imgProps.current.bounds.height) * -1);

    setLeft(left);
    setTop(top);
  }, []);
```

### PR DESCRIPTION
**Pull request by MaxdAyala**
Test code showing how to remove the exception in Firefox.

### CODE REVIEW

File: .github/workflows/test-ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Run unit tests on ubuntu-latest
```
Comment (laurenashpole) [inline .github/workflows/test-ubuntu.yml:1]: The release action for this repo runs tests on `ubuntu-latest`, `macos-latest`, `windows-latest` so a separate file isn't necessary.

File: src/InnerImageZoom/InnerImageZoom.js [hunk 2]
```diff
@@ -100,12 +100,10 @@ const InnerImageZoom = ({
...
   const handleDragStart = (e) => {
-    imgProps.current.offsets = getOffsets(
-      e.pageX || e.changedTouches[0].pageX,
-      e.pageY || e.changedTouches[0].pageY,
-      zoomImg.current.offsetLeft,
-      zoomImg.current.offsetTop
-    );
+    const pageX = typeof e.pageX === 'number' ? e.pageX : e.changedTouches[0].pageX;
+    const pageY = typeof e.pageY === 'number' ? e.pageY : e.changedTouches[0].pageY;
+    //console.log(`Start pageX: ${pageX} pageY: ${pageY}`);
```
Comment (laurenashpole) [inline src/InnerImageZoom/InnerImageZoom.js:105]: This check is a good idea! The commented out code above could probably be removed.

File: src/InnerImageZoom/InnerImageZoom.js [hunk 3]
```diff
@@ -119,8 +117,11 @@ const InnerImageZoom = ({
 
   const handleDragMove = useCallback((e) => {
     e.stopPropagation();
-    let left = (e.pageX || e.changedTouches[0].pageX) - imgProps.current.offsets.x;
-    let top = (e.pageY || e.changedTouches[0].pageY) - imgProps.current.offsets.y;
+    const pageX = typeof e.pageX === 'number' ? e.pageX : e.changedTouches[0].pageX;
+    const pageY = typeof e.pageY === 'number' ? e.pageY : e.changedTouches[0].pageY;
+    //console.log(`Move pageX: ${pageX} pageY: ${pageY}`);
```
Comment (laurenashpole) [inline src/InnerImageZoom/InnerImageZoom.js:122]: This line could also probably be deleted.

**Review by laurenashpole — CHANGES_REQUESTED** [review]
Thanks for submitting this PR and sorry for the delay reviewing! I saw your other comments and realized I'd forgotten to come back to these. I try to do as much QA as I can for every change so I get a little backed up.

This is looking good! I just left a couple comments about extraneous code.

File: src/InnerImageZoom/InnerImageZoom.js (hunk 2 above)
Comment (MaxdAyala) [reply src/InnerImageZoom/InnerImageZoom.js]: Yes, the console.log was just for verifiying the bug and is not needed.

File: src/InnerImageZoom/InnerImageZoom.js (hunk 3 above)
Comment (MaxdAyala) [reply src/InnerImageZoom/InnerImageZoom.js]: Yes, not needed.

**Comment by MaxdAyala** [comment]
Ok, I've made the changes and checks seem to have passed ok.

**Pull request merged**